weaviate/embedded.py

```diff
@@ -188,6 +199,41 @@ def check_supported_platform() -> None:
                 f"you want this: https://github.com/weaviate/weaviate/issues/3315"
             )
 
+    def stop(self) -> None:
+        if self.process is not None:
+            try:
+                self.process.terminate()
+                self.process.wait()
+            except ProcessLookupError:
+                print(
+                    f"""Tried to stop embedded weaviate process {self.process.pid}. Process was not found. So not doing
+                    anything"""
+                )
+            self.process = None
+
+    def ensure_running(self) -> None:
+        if self.is_listening() is False:
+            print(
+                f"Embedded weaviate wasn't listening on ports http:{self.options.port} & grpc:{self.options.grpc_port}, so starting embedded weaviate again"
+            )
+            self.start()
+
+    @abstractmethod
+    def start(self) -> None:
+        raise NotImplementedError()
+
+
+class EmbeddedDB(_EmbeddedBase):
```

Comment:
why are there now two classes?

Reply:
To change the behaviour when connecting to an already existing Weaviate, a breaking change of embedded would happen due to the need to throw an exception, whereas previously it would silently connect to local

By having two classes, the v3 -> v4 migration won't include this breaking change of embedded's behaviour and `EmbeddedDB` will be coupled to the v3 client

Reply:
ahhh, makes sense. Could you rename it to EmbeddedV3/V4? Then it is very clear